cpg-core/src/main/java/de/fraunhofer/aisec/cpg/graph/statements/expressions/ArraySubscriptionExpression.java

```diff
@@ -119,4 +120,10 @@ public int hashCode() {
   public Expression getBase() {
     return this.arrayExpression;
   }
+
+  @Nullable
+  @Override
+  public String getOperatorCode() {
```

Comment:
Why is this necessary or was there a specific design idea behind this change. Also, why does it have a base, I would only expect these to exist if they inherited from BinaryOperator or MemberExpression.

Reply:
Ah no, we made it an implementation of `HasBase`, not so sure about the `getOperatorCode` though

Reply:
the array subscription expression always implemented `HasBase`, which makes sense, since it operates on a specific base (which can be check for null for example). I decided to extend `HasBase` with the `operatorCode`, which was already part of the BinaryOperator and the MemberExpression, to differentiate *how* one access the base.

Reply:
Ok, I like the thought behind it. It only extends the things we can do with the node type so its a good change.